Symbol and footprint for ESP-WROOM-32 module

russss opened the pull request:
As per the datasheet [here](https://espressif.com/sites/default/files/documentation/esp_wroom_32_datasheet_en.pdf).

The ESP32 is quite flexible in that it can remap most alternate functions onto any of the pins. So I've decided to only label the significant pins on the symbol.

The datasheet is suprisingly good but it inconsistently refers to the exposed pad as pin 1 on the diagram (fig 1) and pin 39 in the table (table 2). I've gone with 39.

### Symbol
![image](https://cloud.githubusercontent.com/assets/45057/25799877/c32d9376-33de-11e7-9f3d-482e09d3ab4c.png)

### Footprint
![image](https://cloud.githubusercontent.com/assets/45057/25799910/f3e140da-33de-11e7-9144-0f961c2453fc.png)

adamgreig commented on lib/module/esp-wroom-32.yaml:
```diff
@@ -0,0 +1,45 @@
+datasheet: https://espressif.com/en/products/hardware/esp-wroom-32/overview
```
I'd rather link directly to the datasheet here https://espressif.com/sites/default/files/documentation/esp_wroom_32_datasheet_en.pdf

adamgreig commented on lib/module/esp-wroom-32.yaml:
```diff
@@ -0,0 +1,45 @@
+  - - [EN, 3, in]
+    - [NC, 32, nc]
+  - - [TXD-GPIO1, 35, bidi]
+    - [RXD-GPIO3, 34, bidi]
```
Do you think these should be TXD0 and RXD0 instead of just TXD and RXD, as per the datasheet?

adamgreig commented on lib/module/esp-wroom-32.yaml:
```diff
@@ -0,0 +1,45 @@
+name: ESP-WROOM-32
+ordercodes: []
+pins:
+- - - [VCC, 2, pwrin]
```
Datasheet calls this 3V3 instead of VCC.

russss commented:
Added those changes and rebased.

There doesn't seem to be a canonical footprint anywhere (there are [hardware design guidelines here](http://espressif.com/sites/default/files/documentation/esp32_hardware_design_guidelines_en.pdf) but they don't have a footprint either).

adamgreig commented:
The symbol looks fine to me now, thanks.

On the footprint..

* The F.Fab lines should be 0.01mm thick rather than 0.15mm, as per [the footprint README](https://github.com/adamgreig/agg-kicad/blob/master/agg.pretty/README.md), though this should probably be in the check_mod.py script really
* I forgot previously but we try to avoid silk screen over pads, since it usually won't manufacture and just have to be removed anyway. Could you just remove the silk lines where they go over the pads, leaving stubs on either side? The fab lines can stay as they are.
* The ref and value should go above and below the courtyard rather than inside the part, centered horizontally
* The courtyard should have 0.25mm clearance but currently has 0.1mm on the top and bottom, and 0.05mm on the right. The clearance is from the nearest copper or part outline (here the thin Fab layer is generally easier to measure off and should reflect the part extremities).
